contracts/light_client/src/lib.rs

```diff
@@ -239,9 +269,14 @@ impl LightClient {
         self.next_epoch = from_next_epoch;
 
         if from_next_epoch {
-            let mut next_epoch = &mut self.epochs[(self.current_epoch_index + 2) % NUM_OF_EPOCHS];
+            let epoch_idx = (self.current_epoch_index + 2) % NUM_OF_EPOCHS;
+            let mut next_epoch = &mut self.epochs.iter().nth(epoch_idx).unwrap();
```

Comment:
can &mut be removed?

Reply:
it can, with some other changes. done now.